testutil/inject_validator.go

```diff
@@ -0,0 +1,568 @@
+package testutil
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/linkerd/linkerd2/pkg/k8s"
+)
+
+const enabled = "true"
+const proxyContainerName = "linkerd-proxy"
+const initContainerName = "linkerd-init"
+const debugContainerName = "linkerd-debug"
+
+// InjectValidator is used as a helper to generate
+// correct injector flags and annotations and verify
+// injected pods
+type InjectValidator struct {
```

Comment:
why not use `proxyOptions` here, and re-use the `overrideConfigs` here instead of having a `getFlagsAndAnnotations`? If `proxyOptions` does not contain all fields, maybe that could be a substruct? :thinking:

Reply:
interesting :thinking:

Reply:
Hmm yeah it could be done but I think it clearly serves a different purpose and apart from that `overrideConfigs` takes `values *linkerd2.Values`, so I kind of need to come up with that out of somewhere. Not sure whether it makes much sense to try and fit that here...

Reply:
I was expecting you to pass a `nil` there (as its not used much). But this could be unecessary work and hence resolving this suggestion.